autoload/clap/provider/yanks.vim

```diff
@@ -0,0 +1,70 @@
+" Author: Ratheesh S <ratheeshreddy@gmail.com>
+" Description: List the recently yanked/deleted lines
+" Based on : https://github.com/sgur/ctrlp-extensions.vim
+
+let s:save_cpo = &cpo
+set cpo&vim
+
+let s:yanklist = []
+let s:yanks = {}
+
+function! s:uniq(list) abort
+    let seen = []
+    for i in range(len(a:list))
+        if index(seen, a:list[i]) == -1
+            call add(seen, a:list[i])
+        endif
+    endfor
+    return seen
+endfunction
+
+function! s:remove_duplicated(list, entry) abort
+    let dup = index(a:list, a:entry, 0)
+    if dup > -1
+        call remove(a:list, dup)
+    endif
+endfunction
+
+function! s:yanks_collect() abort
+    let yankstr = getreg('"')
+
+    if !empty(s:yanklist) && yankstr == s:yanklist[0]
+        return
+    endif
+
+    call s:remove_duplicated(s:yanklist, yankstr)
+    call insert(s:yanklist, yankstr)
+endfunction
+
+function! clap#provider#yanks#yanks_init() abort
+    augroup clap_yanklist
+        autocmd!
+        autocmd TextYankPost * call s:yanks_collect()
+    augroup END
+
+    " collect the data from default register
+    call s:yanks_collect()
+endfunction
+
+function! s:yanks.source() abort
+    return s:yanklist
+endfunction
+
+function! s:yanks_sink(selected) abort
+    call setreg('"', a:selected)
+    normal! ""p
+endfunction
+
+let s:yanks.sink = function('s:yanks_sink')
+
+function! s:yanks.on_enter() abort
```

Comment:
The `on_enter` hook is unccessary as the collection is done via the TextYankPost autocmd already. It's often used in this plugin to initialize the filetype of display buffer to get the syntax highlight.

Reply:
`on_enter` hook is removed